src/refactor_nrepl/middleware.clj

```diff
@@ -74,65 +65,112 @@
       )))
 
+(def ^:private resolve-missing
+  (delay
+   (require-and-resolve 'refactor-nrepl.ns.resolve-missing/resolve-missing)))
+
 (defn resolve-missing-reply [{:keys [transport] :as msg}]
-  (reply transport msg :candidates (resolve-missing msg) :status :done))
+  (reply transport msg :candidates (@resolve-missing msg) :status :done))
+
+(def ^:private find-symbol
+  (delay
+   (require-and-resolve 'refactor-nrepl.find.find-symbol/find-symbol)))
 
 (defn- find-symbol-reply [{:keys [transport] :as msg}]
   (config/with-config msg
     (with-errors-being-passed-on transport msg
-      (let [occurrences (find-symbol msg)]
+      (let [occurrences (@find-symbol msg)]
         (doseq [occurrence occurrences
                 :let [response (serialize-response msg occurrence)]]
           (transport/send transport
                           (response-for msg :occurrence response)))
         (transport/send transport (response-for msg :count (count occurrences)
                                                 :status :done))))))
 
+(def ^:private artifact-list
+  (delay (require-and-resolve 'refactor-nrepl.artifacts/artifact-list)))
+
+(def ^:private artifact-versions
+  (delay (require-and-resolve 'refactor-nrepl.artifacts/artifact-versions)))
+
+(def ^:private hotload-dependency
+  (delay (require-and-resolve 'refactor-nrepl.artifacts/hotload-dependency)))
+
 (defn- artifact-list-reply [{:keys [transport] :as msg}]
-  (reply transport msg :artifacts (artifact-list msg) :status :done))
+  (reply transport msg :artifacts (@artifact-list msg) :status :done))
 
 (defn- artifact-versions-reply [{:keys [transport] :as msg}]
-  (reply transport msg :versions (artifact-versions msg) :status :done))
+  (reply transport msg :versions (@artifact-versions msg) :status :done))
 
 (defn- hotload-dependency-reply [{:keys [transport] :as msg}]
-  (reply transport msg :status :done :dependency (hotload-dependency msg)))
+  (reply transport msg :status :done :dependency (@hotload-dependency msg)))
+
+(def ^:private clean-ns
+  (delay
+   (require-and-resolve 'refactor-nrepl.ns.clean-ns/clean-ns)))
+
+(def ^:private pprint-ns
+  (delay
+   (require-and-resolve 'refactor-nrepl.ns.pprint/pprint-ns)))
 
 (defn- clean-ns-reply [{:keys [transport path] :as msg}]
-  (reply transport msg :ns (some-> msg clean-ns pprint-ns) :status :done))
+  (reply transport msg :ns (some-> msg @clean-ns @pprint-ns) :status :done))
+
+(def ^:private find-used-locals
+  (delay
+   (require-and-resolve 'refactor-nrepl.find.find-locals/find-used-locals)))
 
 (defn- find-used-locals-reply [{:keys [transport] :as msg}]
-  (reply transport msg :used-locals (find-used-locals msg)))
+  (reply transport msg :used-locals (@find-used-locals msg)))
 
 (defn- version-reply [{:keys [transport] :as msg}]
   (reply transport msg :status :done :version (core/version)))
 
+(def ^:private warm-ast-cache
+  (delay
+   (require-and-resolve 'refactor.nrepl.analyzer/warm-ast-cache)))
```

Comment:
I think there is typo error.

`refactor.nrepl.analzyer` => `refactor-nrepl.analzyer`